control/plugin/cpolicy/node.go

```diff
@@ -71,6 +71,56 @@ func NewPolicyNode() *ConfigPolicyNode {
 	}
 }
 
+func (c *ConfigPolicyNode) CopyRules() ([]Rule, error) {
+	rules := make([]Rule, len(c.rules))
+	i := 0
+	for _, rule := range c.rules {
```

Comment:
I think this would be clearer as `for i := 0; i < len(c.rules); i++ { rule := c.rules[i];...`.